esti/catalog_export_test.go

```diff
@@ -0,0 +1,418 @@
+          aws_secret_access_key: "{{ .SecretAccessKey }}"
+          aws_region: us-east-1
+        table_source: '{{ .TableDescriptorPath }}'
+        catalog:
+          db_name: "{{ .GlueDB }}"
+          table_input:
+            StorageDescriptor: 
+              InputFormat: "org.apache.hadoop.hive.ql.io.SymlinkTextInputFormat"
+              OutputFormat: "org.apache.hadoop.hive.ql.io.HiveIgnoreKeyTextOutputFormat"
+              SerdeInfo:
+                SerializationLibrary: "org.apache.hadoop.hive.serde2.OpenCSVSerde"
+                Parameters:
+                  separatorChar: ","
+            Parameters: 
+              classification: "csv"
+              "skip.header.line.count": "1"`
+
+const symlinkExporterScript = `
+local exporter = require("lakefs/catalogexport/symlink_exporter")
+local aws = require("aws")
+local table_path = args.table_source
+local s3 = aws.s3_client(args.aws.aws_access_key_id, args.aws.aws_secret_access_key, args.aws.aws_region)
+exporter.export_s3(s3, table_path, action, {debug=true})
+`
+
+const symlinkExporterAction = `
+name: Symlink S3 Exporter
+on:
+  post-commit:
+    branches: ["{{ .Branch }}*"]
+hooks:
+  - id: symlink_exporter
+    type: lua
+    properties:
+      script_path: "{{ .SymlinkScriptPath }}"
+      args:
+        aws:
+          aws_access_key_id: "{{ .AccessKeyId }}"
+          aws_secret_access_key: "{{ .SecretAccessKey }}"
+          aws_region: us-east-1
+        table_source: '{{ .TableDescriptorPath }}'`
+
+func renderTplAsStr(t *testing.T, tplData any, name, content string) string {
+	t.Helper()
+	tpl, err := template.New(name).Parse(content)
+	require.NoError(t, err, "rendering template")
+	var doc bytes.Buffer
+	err = tpl.Execute(&doc, tplData)
+	require.NoError(t, err)
+	return doc.String()
+}
+
+func setupGlueClient(ctx context.Context, accessKeyID, secretAccessKey string) (*glue.Client, error) {
+	cfg, err := config.LoadDefaultConfig(ctx,
+		config.WithRegion("us-east-1"),
+		config.WithCredentialsProvider(credentials.NewStaticCredentialsProvider(accessKeyID, secretAccessKey, "")),
+	)
+	if err != nil {
+		return nil, err
+	}
+	return glue.NewFromConfig(cfg), nil
+}
+
+func uploadAndCommitObjects(t *testing.T, ctx context.Context, repo, branch string, objectsGroups ...map[string]string) *apigen.Commit {
+	t.Helper()
+	for _, objects := range objectsGroups {
+		for path, obj := range objects {
+			resp, err := uploadContent(ctx, repo, branch, path, obj)
+			require.NoError(t, err)
+			require.Equal(t, http.StatusCreated, resp.StatusCode())
+		}
+	}
+
+	commitResp, err := client.CommitWithResponse(ctx, repo, mainBranch, &apigen.CommitParams{}, apigen.CommitJSONRequestBody{
+		Message: "Table Data",
+	})
+	require.NoErrorf(t, err, "failed commiting uploaded objects to lakefs://%s/%s", repo, branch)
+	require.Equal(t, http.StatusCreated, commitResp.StatusCode())
+	return commitResp.JSON201
+}
+func genCSVData(cols []string, n int) string {
```

Comment:
```suggestion

func genCSVData(cols []string, n int) string {
```

Reply:
Please consider using the standard [encoding/csv](https://pkg.go.dev/encoding/csv) package.  Encoding CSVs like this is incredibly brittle.

Reply:
100% done!